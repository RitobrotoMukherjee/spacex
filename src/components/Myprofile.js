import '../css/Missions.css';
import '../css/Profile.css';
import { useSelector } from 'react-redux';
import ProfileTable from './Profile-components/ProfileTables';

const MyProfile = () => {
  const { Rockets } = useSelector((state) => state.rockets);
  const { missionList } = useSelector((state) => state.missions);

  const reservedRockets = Rockets.filter(({ reserved }) => reserved);
  const joinedMissions = missionList.filter(({ reserved }) => reserved);

  return (
    <section className="mission-container">
      <div className="profile-holder">
        <ProfileTable dataList={reservedRockets} header="My Rockets" type="rockets" />
        <ProfileTable dataList={joinedMissions} header="My Missions" type="missions" />
      </div>
    </section>
  );
};

export default MyProfile;
