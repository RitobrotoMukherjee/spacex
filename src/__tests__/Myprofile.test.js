import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/configureStore';
import Myprofile from '../components/Myprofile';

it('Myprofile component snapshot testing', () => {
  const myprofile = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <Myprofile />
        </BrowserRouter>
      </Provider>,
    )
    .toJSON();
  expect(myprofile).toMatchSnapshot();
});
