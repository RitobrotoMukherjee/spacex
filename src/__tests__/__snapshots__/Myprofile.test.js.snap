// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Myprofile component snapshot testing 1`] = `
<section
  className="mission-container"
>
  <div
    className="profile-holder"
  >
    <div
      className="Flex"
    >
      <h2
        className="profile-heading table-heading"
      >
        My Rockets
      </h2>
      <table
        className="profile-table"
      >
        <tbody />
      </table>
    </div>
    <div
      className="Flex"
    >
      <h2
        className="profile-heading table-heading"
      >
        My Missions
      </h2>
      <table
        className="profile-table"
      >
        <tbody />
      </table>
    </div>
  </div>
</section>
`;
